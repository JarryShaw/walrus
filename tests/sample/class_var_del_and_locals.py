x = 'outer'


class A:
    print(x := 666)  # new
    print(x)
    print(locals().get('x'))
    print(x := 777)  # update
    print(x)
    print(locals().get('x'))
    del x  # delete
    print(locals().get('x'))
    print(x := 888)  # recreate
    print(x)
    print(locals().get('x'))


print(x)
