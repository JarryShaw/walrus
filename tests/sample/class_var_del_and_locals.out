666
666
666
777
777
777
None
888
888
888
outer
